test/functorch/test_aotdispatch.py

```diff
@@ -1016,6 +1016,34 @@ def inp_callable(req_grad):
         self.verify_aot_autograd(f, partial(inp_callable, req_grad=False), test_mutation=True)
         self.verify_aot_autograd(f, partial(inp_callable, req_grad=True), test_mutation=True)
 
+    @unittest.skipIf(not torch.cuda.is_available(), "CUDA is unavailable")
+    def test_mem_leak_from_save_for_bw(self):
+        # See a full diagnosis at this issue: https://github.com/pytorch/pytorch/issues/94990
+        # Note [Detaching saved tensors in AOTAutograd]
+        # This program creates a ref-cycle. Long term, we should fix this ref cycle
+        # (since it can arise, naturally albeit rarely, from uses of autograd.Function).
+        # But AOTAutograd makes it more likely to show up from tracing user programs,
+        # so we deal with it by manually detaching the tensors that we save for backward.
+        # This is completely wrong and would give wrong results if we were to do double backward.
+        # Fortunately today, double backward is explicitly banned in AOTAutograd.
```

Comment:
would we obviously trigger this issue if we tried to run double backwards?

Reply:
just checking- you're asking if whether or not we will fail loudly with a nice error message if you try to run double backwards today (and not e.g. silently get wrong results / get a bad error message)?

Jeffrey did some work a few weeks ago to make sure we get a nice error message. I actually just updated this PR to make sure that we still get the nice error (there's an expect test that asserts we get a good error message today).

Reply:
Oh, you're wondering if this ref cycle can show up more commonly if you're doing double backward with an `autograd.Function`?

I'm not 100% sure, but I'm leaning towards no - only this specific usage of autograd.Function will cause the mem leak (regardless of whether you're doing single or double backward).